src/Graph.cpp

```diff
@@ -240,6 +243,58 @@ void Graph::loadFromEdgeListFile(string fin, string graphName, Graph& g, bool no
     g.initConnectedComponents();
 }
 
+void Graph::connected(uint node1, uint node2, vector<vector<ushort>> edgeList, unsigned int i,Graph& g){
+	stringstream errorMsg;
+#ifndef NO_ADJ_MATRIX
+	// the below should be abstracted into a function, eg g.connected(node1, node2)
+	// Inside that function is the only place you'd need to know if adjMatrix exists or not.
+    if(g.adjMatrix[node1][node2] || g.adjMatrix[node2][node1]){
+		//errorMsg << "duplicate edges not allowed (in either direction), node numbers are " << node1 << " " << node2 << '\n';
+	    unordered_map<ushort,string> index2name = g.getIndexToNodeNameMap();
+	    errorMsg << "In graph[" << g.name << "]: duplicate edges not allowed (in either direction), node names are " << index2name[node1] << " " << index2name[node2] << '\n';
+		throw runtime_error(errorMsg.str().c_str());
+    }
+    if(node1 == node2) {
+		errorMsg << "self-loops not allowed, node number " << node1 << '\n';
+        throw runtime_error(errorMsg.str().c_str());
+    }
+
+	// These should be abstracted into g.connect(node1, node2) to cause these two nodes
+	// to get an edge between them.
+	// Note that when WEIGHTED is on, the adjacency matrix contains full integers, not just bits.
+	#ifdef WEIGHTED
+		g.adjMatrix[node1][node2] = g.adjMatrix[node2][node1] = edgeList[i][2];
```

Comment:
`edgeList[i][2]` is the weight in case of "edge list format" (its the third column), so instead just pass the weight in the funtion, and replace here. This way we can avoid passing the `edgeList` to the function.